daemon/container_operations.go

```diff
@@ -73,6 +73,9 @@ func (daemon *Daemon) buildSandboxOptions(cfg *config.Config, container *contain
 	} else if len(cfg.DNSOptions) > 0 {
 		sboxOptions = append(sboxOptions, libnetwork.OptionDNSOptions(cfg.DNSOptions))
 	}
+	if cfg.DNSNoProxy {
+		sboxOptions = append(sboxOptions, libnetwork.OptionDNSNoProxy())
```

Comment:
> I can understand if you would want it to be consistent with how daemon config reload does not affect the DNS configuration of containers until the containers are restarted, though.

Let's go with that. Also...
- there's a list of the [things reload does](https://docs.docker.com/reference/cli/dockerd/#configuration-reload-behavior), I don't know how complete it is, but of course this isn't on it.
- the flag is mainly intended for emergency use, in case this change causes problems for Windows containers. But, daemon reload doesn't seem particularly useable on Windows, I'm not sure it works.
- if we find out that people need to change the setting on reload, or per-container, we can add support for that later. I think there's a fair chance nobody will need it.